testkit/src/main/java/org/apache/calcite/test/SqlOperatorTest.java

```diff
@@ -6965,8 +6965,16 @@ void checkRegexpExtract(SqlOperatorFixture f0, FunctionAlias functionAlias) {
         isWithin(9.0, 0.000001));
     f.checkScalarApprox("log(cast(10e-3 as real), 10)", "DOUBLE NOT NULL",
         isWithin(-2.0, 0.000001));
+    f.checkScalarApprox("log(10)", "DOUBLE NOT NULL",
+        isWithin(2.302585092994046, 0.000001));
     f.checkNull("log(cast(null as real), 10)");
     f.checkNull("log(10, cast(null as real))");
+    f.checkFails("log(0)",
+        "Cannot take logarithm of zero or negative number", true);
+    f.checkFails("log(0, 64)",
```

Comment:
The error message could probably be more descriptive, showing the value which is illegal.
Also, in one of these two cases it's not the number which is illegal, but the base.

Reply:
In fact, in LOG(numeric1 [, base ]) these two parameters cannot both be 0 or negative, otherwise errors will be reported in postgres and bigqeury.